socallinuxexpo/scale-network
# [READY] Resolve some open Ansible issues

## Description of PR
Fixes #207
Fixes #226
Fixes #232

This PR includes fixes for 3 open issues as well as some general housekeeping within the Ansible directory.

## Previous Behavior
- isc-dhcp-server6 service would go OOM during startup or shortly after due to unnecessarily large IPv6 scope space
- IPv4 leases could be renewed for up to 15 minutes on a 5 minute lease, preventing APs from getting timely DHCP related configuration option updates
- Data files were tab separated instead of comma 
- Python2 was still being installed on servers
- Ansible file layout was unnecessarily cumbersome

## New Behavior
- Greatly reduce the IPv6 DHCP scope sizes to address issue #207 . DHCP scope is now 1024 per subnet for each DHCP server. The service no longer crashes with coredump/OOM conditions.
- Set the IPv4 DHCP max lease time equal to the default lease to address issue #226 . The server has no control over clients requesting renewals within the 5 minutes but will no longer allow them after a full first lease.
- Switched fact files from TSV to CSV to address issue #232 . This includes the lists for servers, routers, aps, and pis.
- Removed the python2 role from the project. It will be deprecated before the next SCaLE.
- Simplified the ansible file layout by removing the embedded etc/ansible directory.
- Added a header row to the AP list

## Tests
Test for Issue #207 
1. bring environment up with vagrant up
2. login to server1 with vagrant ssh server1, elevate to root
3. open file to edit /etc/netplan/50-vagrant.yml
4. add item to addresses list - 2001:470:f325:503::5/64 and save file
5. netplan apply
6. verify 2001:470:f325:503::5/64 attached to enp0s8
7. restart isc-dhcp-server6 with systemctl restart isc-dhcp-server6
8. tail dhcp log with tail -f /var/log/dhcp.log
9. in another terminal log in to server8 with vagrant ssh server8, elevate to root
10. created 200 test dhcp requests, for i in {1..200}; do dhclient -6; done;
11. in server1 terminal verify logs
12. verify isc-dhcp-server6 has not crashed with systemctl status isc-dhcp-server6 and viewing the "Active" item

Test for Issue #226 
1. login to server1 with vagrant ssh server1
2. review /etc/dhcp/dhcpd.conf and note that default-lease-time and max-lease-time now match

Test for Issue #232 
1. view the following files in the facts directory {pi/pilist.csv, aps/aplist.csv, servers/serverlist.csv, routers/routerlist.csv} and note they are all csv now
2. run inventory.py and note data structure populated as expected. ./inventory.py | jq "." | less
